internals/search.py

```diff
@@ -121,19 +121,21 @@ def process_queriable_field(field_name, operator, val_str):
   return promise
 
 
-FIELD_NAME_PATTERN = r'[.a-z_0-9]+'
+TEXT_PATTERN = r'[^":=><! ]+|"[^S]+"'
```

Comment:
Since regex notation is often very difficult to read, it would be good to describe what it does in a less precise but easier to understand english comment.

Reply:
Good suggestion.  I added the comments.